### ISSUE TITLE
False Positives for `redundant_discardable_let`

### ISSUE THREAD

**Issue by SDGGiesbrecht** [realm/SwiftLint#1415]
`let` is necessary when trying to use an explicit type to disambiguate the a function, but this still triggers a warning.

For example, this is invalid syntax and does not compile:
```swift
_: UInt = 3 - 2
```
This is correct:
```swift
let _: UInt = 3 - 2
```
...but SwiftLint complains about the `let` and suggests switching back to the first.

**Comment by jpsim** [comment]
Fixed in 9b4f495507cf5e21cee249c2cab85b7844b6b685, fb9c482683e0ea5258f464bdb42083c72064de07 and e2ecafea5d272aa4583e8a1ea65567217fbf24ad.

**Comment by zef** [comment]
I believe I found another false positive with this rule:

`while let _ = foo()`

I believe Swift requires the let here... Thanks!

**Comment by jpsim** [comment]
Correct, that was fixed in #1672.

**Comment by zef** [comment]
Thank you, sorry I didn't find that despite searching!
